core/src/main/java/eu/fasten/core/data/metadatadb/MetadataDao.java

```diff
@@ -693,4 +695,181 @@ public void batchInsertEdges(List<EdgesRecord> edges) {
+                                        .and(pv.CREATED_AT.equal(timestamp))
+                                )
+                        )
+                        .fetch();
+
+        logger.debug("Total rows: " + queryResult.size());
+
+        String result = queryResult.formatJSON();
+        return result;
+    }
+
+    /**
+     * Retrieve a call graph for a given a package name and a timestamp.
+     *
+     * @param forge       Forge of the package
+     * @param packageName Name of the package
+     * @param timestamp   Timestamp when package was created
+     * @param transitive  Boolean option to query transitive relationships
+     * @return callGraph    A JSON-serialized RevisionCallGraph
+     */
+    public String getCallGraph(String forge, String packageName, Timestamp timestamp, boolean transitive) {
+
+        Packages p = Packages.PACKAGES.as("p");
+        PackageVersions pv = PackageVersions.PACKAGE_VERSIONS.as("pv");
+        Modules m = Modules.MODULES.as("m");
+
+        Result<Record> queryResult =
+                context
+                        .select(p.ID, p.FORGE, p.PACKAGE_NAME)
+                        .select(pv.VERSION, pv.CREATED_AT)
+                        .select(m.METADATA.as("module_metadata"))
+                        .from(p)
+                        .innerJoin(pv).on(p.ID.eq(pv.PACKAGE_ID))
+                        .innerJoin(m).on(pv.ID.eq(m.PACKAGE_VERSION_ID))
+                        .where(p.FORGE.equalIgnoreCase(forge)
+                                .and(p.PACKAGE_NAME.equalIgnoreCase(packageName)
+                                        .and(pv.CREATED_AT.equal(timestamp))
+                                )
+                        )
+                        .fetch();
+
+        logger.debug("Total rows: " + queryResult.size());
+
+        String result = queryResult.formatJSON();
+        return result;
+    }
+
+    /**
+     * Gets the vulnerabilities in the transitive closure of a package version
+     *
+     * @param forge       Forge of the package
+     * @param packageName Name of the package
+     * @param version     Version of the package
+     * @return vulnerabilities  Paths of revisions, paths of files/compilation units, paths of functions
+     */
+    public String getVulnerabilities(String forge, String packageName, String version) {
+        // FIXME: Query should be implemented as soon as the data become available
+        Result<Record2<Long, String>> queryResult = context
+                .select(Packages.PACKAGES.ID, Packages.PACKAGES.PACKAGE_NAME)
+                .from(Packages.PACKAGES).fetch();
+
+        for (Record r : queryResult) {
+            Long id = r.getValue(Packages.PACKAGES.ID);
+            String pkgName = r.getValue(Packages.PACKAGES.PACKAGE_NAME);
+            logger.debug("id: " + id + "  / name: " + pkgName);
+        }
+        String result = queryResult.formatJSON();
+        logger.debug("Query dummy result: " + result);
```

Comment:
I don't think that code with "dummy" queries should be merged. I understand that data is not available yet, just don't forget to implement it as soon as the data is there :)